backend/coreapp/compilers.py

```diff
@@ -876,12 +876,50 @@ def available_platforms() -> List[Platform]:
     cc='"${COMPILER_DIR}"/gcc -G0 -c -B "${COMPILER_DIR}"/ $COMPILER_FLAGS "$INPUT" -o "$OUTPUT"',
 )
 
-GCC272SN = GCCCompiler(
-    id="gcc2.7.2sn",
+GCC272SN0001 = GCCCompiler(
+    id="gcc2.7.2sn0001",
+    platform=N64,
+    cc='/usr/bin/cpp -E -lang-c -undef -D__GNUC__=2 -Dmips -D__mips__ -D__mips -Dn64 -D__n64__ -D__n64 -D_PSYQ -D__EXTENSIONS__ -D_MIPSEB -D__CHAR_UNSIGNED__ "$INPUT" '
```

Comment:
are the `cc` commands the same across all versions? can this be pulled out into a variable to avoid the duplication?

Reply:
The cpp flags are the same, yeah, I just copied them from 2.8.1 SN. The other commands are slightly different--0001 uses the default G flag (-G8) whereas 0006 specifies 0001. For dealing with C++, I don't think you can pass a flag to CCN64 to force it to be C++ which is why it has to manually call `cpp` and pass it to the C++ compiler instead of the C one.